lib/nats.js

```diff
@@ -1007,7 +1050,7 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
   } else if (this.closed) {
-    throw(new Error(CONN_CLOSED));
+    throw(new NatsError(CONN_CLOSED_MSG, CONN_CLOSED));
```

Comment:
It becomes a switch by typeof or by code. More things to maintain considering that we have to then alias all those types for typescript.